aerogear/aerogear-unifiedpush-server
# Have APNSCertificate change events use a JMS topic instead of CDI event

Related issue: [AEROGEAR-9068](https://issues.jboss.org/browse/AEROGEAR-9068)

## How to test?

1. Build the project ignoring the tests
   ```shell
   mvn clean install -DskipTests
   ```
1. Add a new new instance of UniedPush Server in `docker-compose/docker-compose-v2.1.yaml` 
   ```yaml
     unifiedpushserver_2:
    image: aerogear/unifiedpush-wildfly-plain:2.2.2-SNAPSHOT
    volumes:
       - ./helper:/ups-helper:z
    entrypoint: "/ups-helper/exportKeycloakHost.sh"
    depends_on:
       unifiedpushDB:
         condition: service_healthy
    environment:
        POSTGRES_SERVICE_HOST: ${POSTGRES_SERVICE_HOST}
        POSTGRES_USER: ${POSTGRES_USER}
        POSTGRES_PASSWORD: ${POSTGRES_PASSWORD}
        POSTGRES_SERVICE_PORT: ${POSTGRES_SERVICE_PORT}
        POSTGRES_DATABASE: ${POSTGRES_DATABASE}
        KEYCLOAK_SERVICE_HOST: ${KEYCLOAK_SERVICE_HOST}
        KEYCLOAK_SERVICE_PORT: ${KEYCLOAK_SERVICE_PORT}
    links:
      - unifiedpushDB:unifiedpush
      - artemis:artemis
      - keycloakServer:keycloak
    ports:
      - 9998:8080      
   ```
1. Run the docker compose. It will run 2 instance of the UPS point to the same postgres and ActiveMQ simulating 2 Openshift pods.
   ```shell
   cd docker-compose && docker-compose -f ./docker-compose-v2.1.yaml up
   ```
1. Create a new project and a iOS variant.
1. Edit cert in the iOS variant (you can change it for the same with the previous one) 
1. See the magics happening between the 2 UPS instance
1. You can also check the ActiveMQ in `http://localhost:8161`. (artemis/simetraehcapa)
